util/ksonnet/ksonnet.go

```diff
@@ -1,49 +1,28 @@
 package ksonnet
 
 import (
+	"encoding/json"
 	"fmt"
+	"io/ioutil"
 	"os/exec"
 	"path/filepath"
 	"strconv"
 	"strings"
 
-	"github.com/ksonnet/ksonnet/pkg/app"
-	"github.com/ksonnet/ksonnet/pkg/component"
 	log "github.com/sirupsen/logrus"
-	"github.com/spf13/afero"
+	yaml "gopkg.in/yaml.v2"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 
 	"github.com/argoproj/argo-cd/pkg/apis/application/v1alpha1"
-	"github.com/argoproj/argo-cd/util/config"
 	"github.com/argoproj/argo-cd/util/kube"
 )
 
-// KsonnetApp represents a ksonnet application directory and provides wrapper functionality around
-// the `ks` command.
-type KsonnetApp interface {
```

Comment:
I still prefer keeping around the ksonnet interface, especially for mocking, and consistency with the helm and kustomize implementations. It can have a reduced method list, but theres value in having it defined as an interface.